test/token-service/test_tokenCreateTransaction.js

```diff
@@ -0,0 +1,8187 @@
+        ),
+      );
+
+      expect(autoRenewPeriod).to.equal(
+        await (
+          await mirrorNodeClient.getTokenData(tokenId)
+        ).auto_renew_period,
+      );
+    }
+
+    it("(#1) Creates a token with an auto renew period set to 0 seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 0,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#2) Creates a token with an auto renew period set to -1 seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: -1,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#3) Creates a token with an auto renew period of 9,223,372,036,854,775,807 (`int64` max) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 9223372036854775807n,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#4) Creates a token with an auto renew period of 9,223,372,036,854,775,806 (`int64` max - 1) seconds", async function () {
+      try {
+        const response = await JSONRPCRequest("createToken", {
+          name: "testname",
+          symbol: "testsymbol",
+          treasuryAccountId: process.env.OPERATOR_ACCOUNT_ID,
+          autoRenewPeriod: 9223372036854775806n,
+        });
+        if (response.status === "NOT_IMPLEMENTED") {
+          this.skip();
+        }
+      } catch (err) {
+        assert.equal(err.data.status, "INVALID_RENEWAL_PERIOD");
+        return;
+      }
+
+      assert.fail("Should throw an error");
+    });
+
+    it("(#5) Creates a token with an auto renew period of 9,223,372,036,854,775,808 (`int64` max + 1) seconds", async function () {
```

Comment:
There is the following check in services:
<img width="544" alt="image" src="https://github.com/user-attachments/assets/a8e37d69-b57a-46fe-89fe-d5e306bbec09">
where `NA = Long.MIN_VALUE;` and the autorenew period gets ignored in this test scenario.

Reply:
Yep its the same in JS, the test does not throw because period is getting ignored.

Reply:
This seems like a bug in services code.  It should not be possible to set a value in the SDK that results in services ignoring the value.
If the services code needs to know if a value is or is not set in protobuf, we need to modify PBJ to support proto3 "explicit presence".  Currently an unset value just gets the field-type default (`0` in this case).